<a href="https://colab.research.google.com/github/thegrouch4413/Predictive-Analysis/blob/main/Ecommerce_Purchases_Exercise_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Ecommerce Purchases Exercise

In this Exercise you will work with 'Fake Data' about some purchases done through Amazon! Follow the directions on the cells bellow and try your best to answer the questions and complete the tasks. Many of the tasks can be solved in 'different ways'.

Do not get discouraged if the exercie does not make "Real-World", REMEMBER that all the data is made-up.

Also note that all of these questions can be answered with one line of code.
____
*Import pandas and read in the Ecommerce Purchases csv file and set it to a DataFrame called ecom.*

In [1]:
import pandas as pd

In [2]:
econ = pd.read_csv('https://raw.githubusercontent.com/glopez21/ML-Data/main/EcommercePurchases.csv')

**Check the head of the DataFrame.**

In [14]:
econ.head()

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4  23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4  Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   
4        Brown, Watson and Andrews  6011456623207998       10/25   

   CC Security Code                  CC Provider  \
0               900                 JCB 16 digit   
1               561                   Mastercard   
2               699                 JCB 16 digit   
3               384                     Discover   
4               678  Diners Club / Carte Blanche   

                            Email                                     Job  \
0               pdunlap@yahoo.com  Scientist, product/process development   
1              anthony41@reed.com                       Drilling engineer   
2  amymiller@morales-harrison.com                Customer service manager   
3     brent16@olson-robinson.info                       Drilling engineer   
4     christopherwright@gmail.com                             Fine artist   

        IP Address Language  Purchase Price  
0  149.146.147.205       el           98.14  
1     15.160.41.51       fr           70.73  
2   132.207.160.22       de            0.95  
3     30.250.74.19       es           78.04  
4     24.140.33.94       es           77.82

*How many rows and columns are there?*

In [15]:
econ.shape

(10000, 14)

There are 10,000 rows and 14 columns.

*What is the average Purchase Price?*

In [16]:
econ['Purchase Price'].mean()

50.34730200000025

The average Purchase Price is $50.35

*What were the highest and lowest purchase prices?*

In [17]:
econ['Purchase Price'].max()

99.99

In [18]:
econ['Purchase Price'].min()

0.0

The highest Purchase Price was $99.99.


The lowest Purchase Price was $0.

*How many people have English 'en' as their Language of choice on the website?*

In [19]:
econ[econ['Language']=='en'].count()

Address             1098
Lot                 1098
AM or PM            1098
Browser Info        1098
Company             1098
Credit Card         1098
CC Exp Date         1098
CC Security Code    1098
CC Provider         1098
Email               1098
Job                 1098
IP Address          1098
Language            1098
Purchase Price      1098
dtype: int64

The total number of people who have english as their language of choice is 1,098.

*How many people have the job title of "Lawyer" ?*


In [20]:
econ[econ['Job']=='Lawyer'].count()

Address             30
Lot                 30
AM or PM            30
Browser Info        30
Company             30
Credit Card         30
CC Exp Date         30
CC Security Code    30
CC Provider         30
Email               30
Job                 30
IP Address          30
Language            30
Purchase Price      30
dtype: int64

The total number of people who have Lawyer as their job title is 30.

*How many people made the purchase during the AM and how many people made the purchase during PM ?*

*(Hint: Check out [value_counts()](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.value_counts.html) )*

In [21]:
econ['AM or PM'].value_counts()

PM    5068
AM    4932
Name: AM or PM, dtype: int64

The total number of people who purchased during AM hours is 4,932.

The total number of people who purchased during PM hours is 5,068.

*What are the 5 most common Job Titles?*

In [22]:
econ['Job'].value_counts().head()

Interior and spatial designer    31
Lawyer                           30
Social researcher                28
Purchasing manager               27
Designer, jewellery              27
Name: Job, dtype: int64

The five most common job titles are interior and spatial designer, lawyer, social researcher, purchasing manager and jewelery designer.

*Someone made a purchase that came from Lot: "90 WT" , what was the Purchase Price for this transaction?*

In [23]:
econ[econ['Lot']=='90 WT']['Purchase Price']

513    75.1
Name: Purchase Price, dtype: float64

The purchase price from lot 90 WT was $75.10

*What is the email of the person with the following Credit Card Number: 4926535242672853*

In [24]:
econ[econ['Credit Card']==4926535242672853]['Email']

1234    bondellen@williams-garza.com
Name: Email, dtype: object

The email of the person with the credit card number 4926535242672853 is bondellen@williams-garza.com.

*How many people have American Express as their Credit Card Provider and made a purchase above $95 ?*

In [26]:
econ[(econ['CC Provider']=='American Express')&(econ['Purchase Price']>95)].count()

Address             39
Lot                 39
AM or PM            39
Browser Info        39
Company             39
Credit Card         39
CC Exp Date         39
CC Security Code    39
CC Provider         39
Email               39
Job                 39
IP Address          39
Language            39
Purchase Price      39
dtype: int64

The total number of people who have American Express and the purchase price is 39.

*Hard: How many people have a credit card that expires in 2025?*

In [27]:
econ[econ['CC Exp Date'].apply(lambda x:x[3:])=='25'].count()

Address             1033
Lot                 1033
AM or PM            1033
Browser Info        1033
Company             1033
Credit Card         1033
CC Exp Date         1033
CC Security Code    1033
CC Provider         1033
Email               1033
Job                 1033
IP Address          1033
Language            1033
Purchase Price      1033
dtype: int64

The total number of people who's credit card expires in 2025 is 39.

*Hard: What are the top 5 most popular email providers/hosts (e.g. gmail.com, yahoo.com, etc...)*

In [28]:
econ['Email'].apply(lambda x:x.split('@')[1]).value_counts().head()

hotmail.com     1638
yahoo.com       1616
gmail.com       1605
smith.com         42
williams.com      37
Name: Email, dtype: int64

The top five most popular email providers are hotmail.com, yahoo.com, gmail.com, smith.com and williams.com.

# Cheers!